src/stactools/viirs/fragment.py

```diff
@@ -0,0 +1,38 @@
+import json
+from typing import Any
+
+import pkg_resources
+from pystac import Extent, Link, MediaType, Provider
+
+
+class STACFragments:
+    """Class for accessing collection and asset data."""
+
+    def __init__(self, product: str) -> None:
+        self.product = product
+
+    def load_assets(self) -> None:
+        self.assets = self._load("assets.json")
+
```

Comment:
This side-effect only method feels a little awkward. This may be a good use case for [cache](https://docs.python.org/3/library/functools.html#functools.cache) on `load_assets`, and then call `load_assets` (maybe renamed to `_load_assets`) in `subdataset_dict`.

Reply:
Yeah, not a good look. I was thinking I would not need to load assets when creating a Collection, so I separated it out. But I end up using the assets for Collection creation anyway, which means I need to access the assets whenever I use an instance of `STACFragments` --> moved this into `__init__` (got rid of the standalone `load_assets()`).

I think we are OK without caching since the assets are now loaded once per Item or Collection creation.